### DISCUSSION TITLE
Is it possible somehow to use form-acitons and prerender?

### DISCUSSION THREAD

**Discussion by alexbjorlig** [sveltejs/kit#7535] [Q&A]
We are trying to build our login page with the new form actions.

It works like a charm - but it fails when we try to prerender the app.

Will this never work - do we have to stop using form-actions if we want to prerender?

**Comment by teemingc** [comment]
It will never work. You will have to either disable prerendering for the page or remove the form actions.

**Reply by ngalaiko** [reply]
do you know what is the limitation exactly? because actions work fine without javascript on the client side, so prerendering the html should be possible?

**Reply by teemingc** [reply]
My understanding is that prerendering computes the page contents at build time. However, form actions occur at runtime and use search parameters e.g., `<form action="todo?/add">` which the docs mention are forbidden.
> Accessing [url.searchParams](https://kit.svelte.dev/docs/load#using-url-data-url) during prerendering is forbidden.

https://kit.svelte.dev/docs/page-options#prerender-when-not-to-prerender

Note that I'm just guessing here, and don't know the actual reason.

**Comment by geoffrich** [comment]
More discussion in https://github.com/sveltejs/kit/issues/7586

> the tl;dr is that you can't POST data to a static file. Most webservers will respond to requests for /YOUR_ROUTE with the static file before even letting SvelteKit see the request, which means you need a server to handle that route.

**Reply by alexbjorlig** [reply]
It makes sense that you can't POST data to a static file, but could the form actions not just live on another route?
